### PR TITLE
SASS-9295 | Tasklist endpoint changes for CIS

### PR DESCRIPTION
**Pull request by Shihabur-Atos**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-CHANGE-ME](ADD URL HERE)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/models/tasklist/TaskTitle.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+sealed abstract class TaskTitle(override val entryName: String) extends EnumEntry {
+  override def toString: String = entryName
+}
+
+object TaskTitle extends Enum[TaskTitle] with PlayJsonEnum[TaskTitle] {
+
+  val values: IndexedSeq[TaskTitle] = findValues
+
+  // UK Self-Employment
```
Comment (hmrcvmu) [inline app/models/tasklist/TaskTitle.scala:29]: Is the comment valid here?

File: conf/app.routes [hunk 2]
```diff
@@ -6,3 +6,5 @@ DELETE  /income-tax/nino/:nino/sources/:submissionId        controllers.DeleteCI
 
 POST    /income-tax/nino/:nino/sources                      controllers.CreateUpdateCisDeductionsController.postCISDeductions(nino: String, taxYear: Int)
 
+GET     /:taxYear/tasks/:nino                               controllers.CommonTaskListController.getCommonTaskList(taxYear: Int, nino: String)
```
Comment (hmrcvmu) [inline conf/app.routes:9]: Should we follow the same format as the other routes?
Maybe `GET /income-tax/nino/:nino/tasks controllers.CommonTaskListController.getCommonTaskList(taxYear: Int, nino: String)`

**Review by hmrcvmu — APPROVED** [review]

File: app/services/CommonTaskListService.scala [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      case Right(cis) => cis
+    }.map { cis =>
+      val tasks: Option[TaskListSectionItem] = getTasks(cis, taxYear)
+      TaskListSection(SectionTitle.CISTitle, tasks)
+    }
+  }
+
+  private def getTasks(cisDeductions: AllCISDeductions, taxYear: Int): Option[TaskListSectionItem] = {
+
+    // TODO: these will be links to the new individual CYA pages when they are made
```
Comment (ghousedev) [inline app/services/CommonTaskListService.scala:44]: You can remove this comment, CIS will not need changes and should load the current CYA page with no problems.

File: test/models/taskList/TaskTitleSpec.scala [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.scalatest.matchers.must.Matchers
+import play.api.libs.json.{JsPath, JsSuccess, JsValue, Json}
+
+class TaskTitleSpec extends AnyFreeSpec with Matchers {
+
+  "TaskTitle" - {
+
+    "must contain the correct values" in {
+      TaskTitle.values mustEqual Seq[TaskTitle](
+        cisDeductions,
```
Comment (ghousedev) [inline test/models/taskList/TaskTitleSpec.scala:31]: You can remove the ',' at the end of line 31.

**Review by ghousedev — APPROVED** [review]

**Review by vivekhmrc — APPROVED** [review]

**Pull request merged**